test/unit/Heatmap/GMUHeatmapTileLayerTest.swift

```diff
@@ -0,0 +1,102 @@
+  private var secondTestCoordinate: CLLocationCoordinate2D!
+  private var intensity: Float!
+  private var cameraLatitude: Double!
+  private var cameraLongitude: Double!
+  private var mapsAPIKey: String!
+  private var radius: UInt!
+  private var minimumZoomIntensity: UInt!
+  private var maximumZoomIntensity: UInt!
+  
+  override func setUp() {
+    super.setUp()
+    startPoints = [NSNumber(value: 0.2), NSNumber(value: 1.0)]
+    colorMapSize = 3
+    gradientColor = [
+      UIColor(red: 102.0 / 255.0, green: 225.0 / 255.0, blue: 0, alpha: 1),
+      UIColor(red: 1.0, green: 0, blue: 0, alpha: 1)
+    ]
+    firstTestCoordinate = CLLocationCoordinate2DMake(10.456, 98.122)
+    secondTestCoordinate = CLLocationCoordinate2DMake(10.556, 98.422)
+    intensity = 10.0
+    cameraLatitude = -33.8
+    cameraLongitude = 151.2
+    radius = 20
+    minimumZoomIntensity = 5
+    maximumZoomIntensity = 10
+    //Please provide your GMAP API Key here.
+    mapsAPIKey = ""
+  }
+  
+  override func tearDown() {
+    gradientColor = nil
+    startPoints = nil
+    colorMapSize = nil
+    firstTestCoordinate = nil
+    secondTestCoordinate = nil
+    intensity = nil
+    cameraLatitude = nil
+    cameraLongitude = nil
+    mapsAPIKey = nil
+    radius = nil
+    minimumZoomIntensity = nil
+    maximumZoomIntensity = nil
+    super.tearDown()
+  }
+  
+  func testInitWithValidGradientColorCount() {
+    let heatmapTileLayer = GMUHeatmapTileLayer()
+    heatmapTileLayer.gradient = GMUGradient(colors: gradientColor, startPoints: startPoints, colorMapSize: colorMapSize)
+    XCTAssertEqual(heatmapTileLayer.gradient.colors.count, gradientColor.count)
+  }
+  
+  func testHeatMapTileLayerDataPoints() {
+    //Please provide your GMAP API key, to run the test case.
+    guard let key = mapsAPIKey, !key.isEmpty else { return }
+    GMSServices.provideAPIKey(mapsAPIKey)
+    let heatmapTileLayer = GMUHeatmapTileLayer()
+    heatmapTileLayer.gradient = GMUGradient(colors: gradientColor, startPoints: startPoints, colorMapSize: colorMapSize)
+    heatmapTileLayer.weightedData = [GMUWeightedLatLng(coordinate: firstTestCoordinate, intensity: intensity), GMUWeightedLatLng(coordinate: secondTestCoordinate, intensity: intensity)]
+    let camera = GMSCameraPosition.camera(withLatitude: cameraLatitude, longitude: cameraLongitude, zoom: 4)
+    heatmapTileLayer.map = GMSMapView.map(withFrame: .zero, camera: camera)
+    XCTAssertEqual(heatmapTileLayer.weightedData.count, 2)
+    XCTAssertEqual(heatmapTileLayer.gradient.colors.count, 2)
+    XCTAssertEqual(heatmapTileLayer.radius, radius)
+    XCTAssertEqual(heatmapTileLayer.minimumZoomIntensity, minimumZoomIntensity)
+    XCTAssertEqual(heatmapTileLayer.maximumZoomIntensity, maximumZoomIntensity)
+  }
+  
+  func testTileLayerForMinXLessThanMinusOneWithNotNilUIImage() {
+    let heatmapTileLayer = GMUHeatmapTileLayer()
+    XCTAssertEqual(heatmapTileLayer.tileFor(x: UInt(0.1), y: UInt(0.1), zoom: 0), UIImage())
```

Comment:
The method name of this indicates that the expectation is not nil but the test case is actually checking for a _specific_ value (i.e. UIImage). Should this be modified to `XCTAssertNotNil`?

Reply:
yes, my bad, written not nil as method name but did `XCTAssertEqual`